torch/csrc/jit/codegen/cuda/kernel_ir_builder.h

```diff
@@ -49,6 +49,14 @@ class TORCH_CUDA_CU_API IrBuilder {
     return node;
   }
 
+  //! Marks op as omitted if the ops supports, which will
+  //!  change the behavior of code generation for this op
+  template <class T>
+  void markOmitted(T* op) {
+    const kir::Passkey passkey(kernel_);
+    op->markOmitted(passkey);
+  }
```

Comment:
Why does this need to be part of the KIR builder?

Reply:
I just wanted to mark the broadcasts and carry the marks into `Kernel` object. The options are either marking on the IR or adding another map into `Kernel` object.

I was going for the earlier approach and wanted to make the method private. Also didn't want to break the passkey pattern so builder is the only place left. 

I guess I could also go for `adding another map into kernel object` approach if this looks too heavy.

Reply:
Hmm, do you really need to use the passkey pattern here? I thought it was meant to restrict creation of KIR nodes. Here, it just modifies a node property, so I'm unsure we would need to use the passkey pattern.

Reply:
I guess if I wanted to mark this property on the IR node, I'd need to follow the passkey pattern, since I don't want to make `markOmitted` public, and inserting friends to the IR node would break the protection by having the passkey. 

My understanding is that, roughly:
A `public` section gives access to `selected methods` to `all objects`. (Too wide audience)
A `friend` gives access to `all methods` to `selected objects`. (Too much power given to some audience)
A `passkey` gives access to `selected methods` to `selected objects`. (More or less what we need here.)

So yes we currently apply passkey pattern to the constructor we control creation of nodes. But we might also consider using this pattern for other cases if it makes sense. In the most general extreme it could be another `transform` class with `transformKey` to update the IR property but that'd not be worth investing currently.

But anyways, I could always fallback to not marking this information on IR nodes if that sounds significantly preferable.

Reply:
I see your point, but I thought a public function would be just fine. When constructing KIR nodes, it's also necessary to ensure created objects to be registered in a `Kernel` object, so the passkey interface is useful as it can enforce it. In this case, it just modifies a property of an object, so I doubt it would be much better than just making it a normal public function.

Reply:
Sure. I will use a public method instead.